src/app/jscaip/Rules.ts

```diff
@@ -40,23 +48,30 @@ export abstract class Rules<M extends Move, S extends GameState, L = void> {
         const child: GameNode<M, S> = new GameNode(resultingState as S, MGPOptional.of(node), MGPOptional.of(move));
         return MGPOptional.of(child);
     }
+
     /**
      * Applies a legal move, given the precomputed information `info`
      */
     public abstract applyLegalMove(move: M, state: S, info: L): S;
+
     /**
      * Returns success if the move is legal, with some potentially precomputed data.
      * If the move is illegal, returns a failure with information on why it is illegal
      */
     public abstract isLegal(move: M, state: S): MGPFallible<L>;
 
-    public getInitialNode(): GameNode<M, S> {
+    public getInitialNode(config?: C): GameNode<M, S> {
+        if (config != null && Object.keys(config).length > 0) {
```

Comment:
est-ce que le deuxième check est vraiment utile ? S'il y a pas de keys il ne peut y avoir qu'une seule config (`{}`), donc si tu fais `this.config = config`, c'est du no-op !